game/src/main/java/jake2/game/SubgameEntity.java

```diff
@@ -227,6 +229,7 @@ public SubgameEntity(int i) {
 
     public ExtraSpawnProperties st = new ExtraSpawnProperties();
 
+    public EnumMap<ComponentType, Object> components = new EnumMap<>(ComponentType.class);
```

Comment:
What are your thoughts on a TypeMap for components?

Reply:
Ok now I see a problem with this approach
https://stackoverflow.com/questions/2625546/is-using-the-class-instance-as-a-map-key-a-best-practice

Reply:
thanks for the hint, I liked it